### PR TITLE
added_Ñsolution_Docker

### PR DESCRIPTION
**Pull request by OlegShvets1**
- added DockerFile, docker-compose.yml, .env
- modified application.properties, liquibase.properties, pom.xml

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+MYSQL_ROOT_PASSWORD=12345678
+MYSQLDB_PASSWORD=12345678
+
+MYSQLDB_LOCAL_PORT=3307
+MYSQLDB_DOCKER_PORT=3306
+
+SPRING_LOCAL_PORT=8081
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (okuzan) [inline .env]: the point of .env file is that it shouldn't be committed - it contains secrets. Not in your case, but that's a good practice to learn.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+# Final stage
+FROM openjdk:17-jdk-alpine
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
+EXPOSE 8080
```
Comment (okuzan) [inline Dockerfile]: don't forget that each file should end with exactly one empty line

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+          "spring.datasource.url" : "jdbc:mysql://mysql:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+          "spring.datasource.username" : "$MYSQLDB_USER",
+          "spring.datasource.password" : "$MYSQL_ROOT_PASSWORD",
+          "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect"
+        }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (okuzan) [inline docker-compose.yml]:
```suggestion
      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
      
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    env_file: ./.env
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+          "spring.datasource.url" : "jdbc:mysql://mysql:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+          "spring.datasource.username" : "$MYSQLDB_USER",
+          "spring.datasource.password" : "$MYSQL_ROOT_PASSWORD",
+          "spring.jpa.properties.hibernate.dialect" : "org.hibernate.dialect.MySQLDialect"
```
Comment (okuzan) [inline docker-compose.yml]: dialect is probably redundant

**Review by okuzan — CHANGES_REQUESTED** [review]

File: .env
Comment (olekskov) [inline .env:1]: env file is still in PR. Please remove it and add to .gitignore

**Review by olekskov — CHANGES_REQUESTED** [review]

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**